code/modules/mob/living/living_healthscan.dm

```diff
@@ -0,0 +1,655 @@
+	)
+
+	var/internal_bleeding = FALSE
+
+	if(data_detail_level)
+		detail_level = data_detail_level
+		data["detail_level"] = detail_level
+	else
+		data["detail_level"] = detail_level
+
+	// chems data
+	data["has_unknown_chemicals"] = FALSE
+	var/list/chemicals_lists = list()
+	for(var/datum/reagent/reagent in target_mob.reagents.reagent_list)
+		if(!(reagent.flags & REAGENT_SCANNABLE) && detail_level == DETAIL_LEVEL_HEALTHANALYSER)
+			data["has_unknown_chemicals"] = TRUE
+			continue
+		chemicals_lists["[reagent.name]"] = list(
+			"name" = reagent.name,
+			"amount" = round(reagent.volume, 0.1),
+			"od" = reagent.overdose != 0 && reagent.volume > reagent.overdose && !(reagent.flags & REAGENT_CANNOT_OVERDOSE),
+			"dangerous" = reagent.overdose != 0 && reagent.volume > reagent.overdose && !(reagent.flags & REAGENT_CANNOT_OVERDOSE) || istype(reagent, /datum/reagent/toxin),
+			"colour" = reagent.color
+		)
+	data["has_chemicals"] = length(target_mob.reagents.reagent_list)
+	data["chemicals_lists"] = chemicals_lists
+
+	var/list/limb_data_lists = list()
+	// human data
+	if(ishuman(target_mob))
+		var/mob/living/carbon/human/human_target_mob = target_mob
+
+		// blood and species
+		var/has_blood = TRUE
+		if(human_target_mob.species.flags & NO_BLOOD)
+			has_blood = FALSE
+		data["has_blood"] = has_blood
+		data["species"] = human_target_mob.species.group
+
+		// permadeadness
+		var/permadead = FALSE
+		if(!human_target_mob.is_revivable())
+			permadead = TRUE
+		else if(!human_target_mob.check_tod() && !isSynth(human_target_mob))
+			permadead = TRUE
+		if(isSynth(target_mob))
+			permadead = FALSE
+
+		data["permadead"] = permadead
+
+		//snowflake :3
+		data["lung_ruptured"] = human_target_mob.is_lung_ruptured()
+
+		//shrapnel, limbs, limb damage, limb statflags, cyber limbs
+		var/core_fracture_detected = FALSE
+		var/unknown_implants = 0
+		for(var/obj/limb/limb in human_target_mob.limbs)
+			var/internal_bleeding_check = FALSE
+			for(var/datum/wound/wound in limb.wounds)
+				if(!istype(wound, /datum/wound/internal_bleeding))
+					continue
+				internal_bleeding = TRUE
+				if(detail_level >= DETAIL_LEVEL_BODYSCAN)
+					internal_bleeding_check = TRUE
+				break
+			if(limb.hidden)
+				unknown_implants++
+			var/implant = FALSE
+			if(length(limb.implants))
+				for(var/I in limb.implants)
+					if(is_type_in_list(I, GLOB.known_implants))
+						continue
+					unknown_implants++
+					implant = TRUE
+
+			var/bleeding_check = FALSE
+			for(var/datum/effects/bleeding/external/E in limb.bleeding_effects_list)
+				bleeding_check = TRUE
+				break
+
+			if((!limb.brute_dam && !limb.burn_dam && !(limb.status & LIMB_DESTROYED)) && !bleeding_check && !internal_bleeding_check && !(implant && detail_level >= DETAIL_LEVEL_BODYSCAN ) && !(limb.status & LIMB_UNCALIBRATED_PROSTHETIC) && !(limb.status & LIMB_BROKEN) && !(limb.status & LIMB_SPLINTED) && !(limb.status & LIMB_SPLINTED_INDESTRUCTIBLE))
```

Comment:
This is a really, really long line - I would try and reformat to be a bit better to read